### PR TITLE
Upgrade Morpho #1

### LINKED ISSUE #1517: Spearbit audit findings
- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/71 [CarryOver From Prev Audit](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22CarryOver+From+Prev+Audit%22) [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/70 [CarryOver From Prev Audit](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22CarryOver+From+Prev+Audit%22) [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/69 [CarryOver From Prev Audit](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22CarryOver+From+Prev+Audit%22) [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/68 [CarryOver From Prev Audit](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22CarryOver+From+Prev+Audit%22) [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/67 [CarryOver From Prev Audit](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22CarryOver+From+Prev+Audit%22) [Severity: High Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+High+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/66 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/65 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/64 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/63 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/62 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/60 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/59 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/58 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/57 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/56 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/55 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/54 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/53

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/52 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/51 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/50 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/49 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/48 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/47 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/46 [Severity: Low Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Low+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/45 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/44 [Severity: Low Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Low+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/43 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/42 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/41 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/40 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/39 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/38 [Severity: Gas Optimization](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Gas+Optimization%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/37 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/36 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/35 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/34 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/33 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/32 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/31 [Severity: Low Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Low+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/30 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/29 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/28 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/27 [Severity: Gas Optimization](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Gas+Optimization%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/26 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/25 [Severity: High Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+High+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/24 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/23 [Severity: Low Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Low+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/22 [Severity: Informational](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Informational%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/21 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/20 [Severity: Gas Optimization](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Gas+Optimization%22)

- [x] https://github.com/spearbit-audits/morpho-novemberAudit/issues/19 [Severity: Medium Risk](https://github.com/spearbit-audits/morpho-novemberAudit/issues?q=is%3Aissue+is%3Aopen+label%3A%22Severity%3A+Medium+Risk%22)

### PR DESCRIPTION
**Pull request by Rubilmax**
# Pull Request

Fixes #1517 

Before merging this PR:
- [x] Merge https://github.com/morpho-dao/morpho-data-structures/pull/108
- [x] Update `lib/morpho-data-structures` submodule

### CODE REVIEW

**Review by pakim249CAL — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

File: src/compound/InterestRatesManager.sol [hunk 1]
```diff
@@ -54,41 +52,40 @@ contract InterestRatesManager is IInterestRatesManager, MorphoStorage {
     function updateP2PIndexes(address _poolToken) external {
         Types.LastPoolIndexes storage poolIndexes = lastPoolIndexes[_poolToken];
```
Comment (MathisGD) [inline src/compound/InterestRatesManager.sol:53]: Misleading, as we use `poolSupplyIndex` and `poolBorrowIndex` for the current ones
```suggestion
        Types.LastPoolIndexes storage lastPoolIndexes = lastPoolIndexes[_poolToken];
```

**Review by MathisGD — APPROVED** [review]
LGTM, optional changes:

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]
The last steps described in the PR seems not to have been done

**Review by MerlinEgalite — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -41,8 +41,9 @@ It also interacts with `RewardsManager`, which manages the underlying pool's rew
 ## Documentation
 
 - [White Paper](https://whitepaper.morpho.xyz)
+- [Yellow Paper](https://yellowpaper.morpho.xyz/)
 - [Morpho Documentation](https://docs.morpho.xyz)
-- Yellow Paper (coming soon)
+- [Morpho Developers Documentation](https://developers.morpho.xyz)
```
Comment (julien-devatom) [inline README.md:46]: You can also add a desc on how to autogen the developers docs, if needed
To serve it locally:
```
forge doc --build --serve --port 4000
```

**Review by MathisGD — DISMISSED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by MathisGD — APPROVED** [review]

**Pull request merged**